### ISSUE TITLE
Qwen2: PAD token = EOS token?

### ISSUE THREAD

**Issue by Qubitium** [QwenLM/Qwen#1080]
### Current Behavior

Why does qwen2 sets pad token equal to eos token? If this is done and padding is used (likely in most training code) the attention mask is set to -1 for all eos/pad tokens. 

There is no way to distinguish eos from pad so everything is attention masked to -1?

Or is qwen2 model/tokenizer code already auto detect first eos and treats as such and only subsequent eos are attention masked to -1?

### Expected Behavior

pad_token != eos_token

**Comment by Qubitium** [comment]
Closed: https://github.com/QwenLM/Qwen/blob/main/tokenization_note.md
